backend/models/sequence.js

```diff
@@ -144,25 +286,70 @@ Sequence.getList = async (account, model, branch, revision, cleanResponse = fals
 };
 
 Sequence.updateSequence = async (account, model, sequenceId, data) => {
-	if (!data || !data.name || !utils.isString(data.name) || data.name === ""  || data.name.length >= 30) {
+	const toUpdate = {};
+	const toSet = {};
+	const toUnset = {};
+
+	if (!data || !sequenceEditSchema.isValidSync(data, { strict: true })) {
 		throw responseCodes.INVALID_ARGUMENTS;
 	}
 
-	const { result } = await db.update(account, sequenceCol(model),
-		{_id: utils.stringToUUID(sequenceId)}, {$set: {name: data.name}});
-	if (result.nModified === 0) {
-		throw responseCodes.SEQUENCE_NOT_FOUND;
+	await Sequence.sequenceExists(account, model, sequenceId);
+
+	if (data.name) {
+		if (!utils.isString(data.name) || data.name === "" || data.name.length >= 30) {
+			throw responseCodes.INVALID_ARGUMENTS;
+		}
+
+		toSet.name = data.name;
+	}
+
+	if (data.revId || data.revId === null || data.frames) {
+		const customSequence = await db.findOne(account, sequenceCol(model),
+			{_id: utils.stringToUUID(sequenceId), customSequence: true});
+
+		if (!customSequence) {
+			throw responseCodes.SEQUENCE_READ_ONLY;
+		}
+
+		if (data.revId) {
+			const history = await History.getHistory(account, model, undefined, data.revId, {_id: 1});
+
+			toSet.rev_id = history._id;
```

Comment:
Can we not do this and just take in `rev_id` instead of revId. it's just add confusion.

Reply:
Also here:
https://github.com/3drepo/3drepo.io/pull/2537/files#diff-ffb4e106db66bb79d9156f73ad54a3a278b6e7f9c41032b5037a2d790432f890R374

We're taking in `revId` as query string here and also POST/PATCH request . but on GET we're returning `rev_id`.

Just make everything consistent.